custom/icds_reports/sqldata/exports/dashboard_usage.py

```diff
@@ -58,206 +56,138 @@ def get_role_from_username(self, username):
+            return True
+        return False
 
     def get_excel_data(self):
         excel_rows = []
         filters = [['Generated at', india_now()]]
 
-        end_date = datetime.datetime.utcnow()
-        start_date = end_date - datetime.timedelta(days=7)
-
         headers = ['Sr.No', 'State/UT Name', 'District Name', 'Block Name', 'Username', 'Level', 'Role',
-                   'Launched?', 'Last Login', 'Logged in the last week?', 'Total', 'Child', 'Pregnant Women',
-                   'Demographics', 'System Usage', 'AWC infrastructure', 'Child Growth Monitoring List',
-                   'ICDS - CAS Monthly Register', 'AWW Performance Report', 'LS Performance Report',
-                   'Take Home Ration']
+                   'Launched?', 'Last Login  Activity ', 'Activity in the last 7 days?']
         serial_count = 0
-        logged_in_user_locations = list(self.user.get_sql_locations(self.domain))
-        if not logged_in_user_locations:
-            self.national_user = True
-
-        loop_counter = 0
-        # Need to fetch all users for a national user
-        while (self.national_user and loop_counter < 1) or len(logged_in_user_locations) > loop_counter:
-            # getting the location types to retrieve for this user location
-
-            location_type_filter = {
-                'aggregation_level': 5
-            }
-            if not self.national_user:
-                user_location = logged_in_user_locations[loop_counter]
-                user_location_type_name = \
-                    self.get_location_id_string_from_location_type(user_location.location_type_name)
-                location_type_filter[user_location_type_name] = user_location.get_id
-            else:
-                user_location_type_name = None
-            for test_location in self.location_test_fields:
-                location_type_filter[test_location] = 0
 
-            all_awc_locations = AwcLocation.objects.filter(**location_type_filter).values(*self.required_fields)
-            # converting the result set to matrix to fetch ancestors for a given location
-            location_matrix, location_ids =\
-                self.convert_rs_to_matrix(all_awc_locations, user_location_type_name)
+        location_type_filter = {
+            'aggregation_level': 5
+        }
+        if not self.national_user:
+            user_location = self.user.get_sql_location(self.domain)
+            user_location_type_name = \
+                self.get_location_id_string_from_location_type(user_location.location_type_name)
+            location_type_filter[user_location_type_name] = user_location.get_id
+        else:
+            user_location_type_name = None
+        for test_location in self.location_test_fields:
+            location_type_filter[test_location] = 0
+
+        all_awc_locations = AwcLocation.objects.filter(**location_type_filter).values(*self.required_fields)
```

Comment:
you can filter just the `aggregation_level=3` you don't need to filter on other level.s